jackson-datatype-problem/src/test/java/org/zalando/problem/IOProblem.java

```diff
@@ -0,0 +1,84 @@
+package org.zalando.problem;
+
+/*
+ * ⁣​
+ * Jackson-datatype-Problem
+ * ⁣⁣
+ * Copyright (C) 2015 - 2016 Zalando SE
+ * ⁣⁣
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ​⁣
+ */
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Optional;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NONE)
+public final class IOProblem extends IOException implements Exceptional {
+
+    private final URI type;
+    private final String title;
+    private final Response.StatusType status;
+    private final Optional<String> detail;
+    private final Optional<URI> instance;
+
+    @JsonCreator
+    public IOProblem(@JsonProperty("type") final URI type,
+            @JsonProperty("title") final String title,
+            @JsonProperty("status") final int status,
+            @JsonProperty("detail") final Optional<String> detail,
+            @JsonProperty("instance") final Optional<URI> instance) {
+        this.type = type;
+        this.title = title;
+        this.status = MoreStatus.fromStatusCode(status);
+        this.detail = detail;
+        this.instance = instance;
+    }
+
+    @Override
+    public URI getType() {
+        return type;
+    }
+
+    @Override
+    public String getTitle() {
+        return title;
+    }
+
+    @Override
+    public Response.StatusType getStatus() {
+        return status;
+    }
+
+    @Override
+    public Optional<String> getDetail() {
+        return detail;
+    }
+
+    @Override
+    public Optional<URI> getInstance() {
+        return instance;
+    }
+
+    @Override
+    public ThrowableProblem getCause() {
```

Comment:
Could `getCause` be a default method in `Exceptional`? Having to override it just to return null is not really intuitive.